cycler.py

```diff
@@ -113,10 +113,26 @@ def __init__(self, left, right=None, op=None):
         Do not use this directly, use `cycler` function instead.
         """
         self._keys = _process_keys(left, right)
-        self._left = copy.deepcopy(left)
-        self._right = copy.deepcopy(right)
+        if isinstance(left, Cycler):
+            self._left = left._shallow_copy()
+        else:
+            self._left = copy.copy(left)
+
+        if isinstance(right, Cycler):
+            self._right = right._shallow_copy()
+        else:
+            self._right = copy.copy(right)
+
         self._op = op
 
+    def _shallow_copy(self):
+        ret = Cycler(None)
```

Comment:
I think we can do this block as `return Cycler(self._left, self._right, self.op)` 

ps, sorry for the  backwards lisp